go/types/map.go

```diff
@@ -218,6 +218,25 @@ func (m Map) IterAll(cb mapIterAllCallback) {
 	})
 }
 
+func (m Map) IterAllRange(start, end Value, includeStart, includeEnd bool, cb mapIterAllCallback) {
```

Comment:
"All" vs "Range" seems contradictory. Can this be `IterRange()`?

Also could the caller just decide whether to include the start or end at runtime? It would make this interface simpler. `IterRange` can - and should document - that the range on either end is inclusive.